### PR TITLE
H-2081: Create embeddings for types

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

To be able to infer types, the ontology types requires some embedding as well.

## 🔍 What does this change?

For all three ontology types:
- Create workflows to create the embeddings
- Allow calling workflow trigger from Rust
- Add endpoint to the graph to insert embeddings
- update embedding when type is created/updated

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

## 🐾 Next steps

- Allow querying for embedding distance (H-2167)

## 🛡 What tests cover this?

A few small tests will be added as soon as querying for the embedding distance is possible

### CODE REVIEW

File: apps/hash-ai-worker-ts/src/activities/shared/embeddings.ts [hunk 1]
```diff
@@ -104,3 +107,87 @@ export const createEntityEmbeddings = async (params: {
...
+}): Promise<{
+  embeddings: {
+    embedding: number[];
+  }[];
+  usage: Usage;
+}> => {
+  // We want to create embeddings for:
+  //   1. The `Title: Description` pair
+  //
+  // We use the last item in the array to store the combined 'all properties' list.
```
Comment (CiaranMn) [inline apps/hash-ai-worker-ts/src/activities/shared/embeddings.ts:122]:
```suggestion
```

File: apps/hash-ai-worker-ts/src/activities/shared/embeddings.ts [hunk 2]
```diff
@@ -104,3 +107,87 @@ export const createEntityEmbeddings = async (params: {
...
```
Comment (CiaranMn) [inline apps/hash-ai-worker-ts/src/activities/shared/embeddings.ts:150]:
```suggestion
```

File: apps/hash-ai-worker-ts/src/activities/shared/embeddings.ts [hunk 3]
```diff
@@ -104,3 +107,87 @@ export const createEntityEmbeddings = async (params: {
...
```
Comment (CiaranMn) [inline apps/hash-ai-worker-ts/src/activities/shared/embeddings.ts:178]:
```suggestion
```

File: apps/hash-ai-worker-ts/src/workflows.ts [hunk 4]
```diff
@@ -70,25 +74,283 @@ export const inferEntities = async (params: InferEntitiesCallerParams) => {
...
+    }
+  | {
+      filter: Filter;
+    }
+);
+
+export const updateDataTypeEmbeddings = async (
+  params: UpdateDataTypeEmbeddingsParams,
+): Promise<CreateEmbeddingResponse.Usage> => {
+  const temporalAxes = {
```
Comment (CiaranMn) [inline apps/hash-ai-worker-ts/src/workflows.ts:93]: Can we use `currentTimeInstantTemporalAxes` from `@local/hash-isomorphic-utils/graph-queries` instead? Saves having to work out what it is.

**Review by CiaranMn — COMMENTED** [review]
Thanks @TimDiekmann! a couple of minor comments on the TS code

File: apps/hash-ai-worker-ts/src/workflows.ts (hunk 4 above)
Comment (TimDiekmann) [reply apps/hash-ai-worker-ts/src/workflows.ts:93]: No, sadly not because this is inside of a workflow and the package depends on the type system package.

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**